### ISSUE TITLE
Microsecond Normalization

### ISSUE THREAD

**Issue by muthu3107** [pascaldekloe/jwt#20]
Hi.
Have configured all timestamps in **microseconds**.
But in library i have seen only Unix Timestamp in seconds is used, have found an option for TimeFunc(), but with that only Timezone can be changed.

Can anyone help me on this? or any workaround for the same?

While debugging,

Time configured as **Microseconds** in token:
![image](https://github.com/pascaldekloe/jwt/assets/71437724/ef328cfc-0eb8-40d0-b507-a7992bd34f46)

_var TimeFunc = time.Now
now := TimeFunc().Unix()_

Current time in **Seconds**:
![image](https://github.com/pascaldekloe/jwt/assets/71437724/913c9b42-f6d0-424a-86b8-c141d6a0de18)

P.s, Apologies if i didnt follow standard way of raising concern

**Comment by pascaldekloe** [comment]
Hey Muthu,

I'm not sure about what you mean by “configured all timestamps in microseconds”.

If you want the "nbf" in microseconds then you can use the standard `UnixMicro` method from Go as follows.

```go
nbf := claims.NotBefore.Time().UnixMicro()
```

**Comment by muthu3107** [comment]
Hey,
Below is how i have generated jwt token,
![image](https://github.com/pascaldekloe/jwt/assets/71437724/29a7e8af-96fb-4c62-9e7a-c3f1e04b06d7)


I encountered an issue while using the default token.valid after parsing, where it throws an error even though the time is valid. During debugging, I discovered that internally, the library's configuration is in seconds. However, I am unsure how to configure it accordingly.
